sdl-core/src/main/scala/io/smartdatalake/util/hive/HiveUtil.scala

```diff
@@ -645,4 +654,81 @@ private[smartdatalake] object HiveUtil extends SmartDataLakeLogger {
+    val otherStats = Seq(Some(TableStatsType.CreatedAt.toString -> metadata.createTime), lastAnalyzedAt.map(v =>  TableStatsType.LastAnalyzedAt.toString -> v), lastAnalyzedColumnsAt.map(v =>  TableStatsType.LastAnalyzedColumnsAt.toString -> v)).flatten.toMap
+    otherStats ++ catalogStats
+  }
+
+  /**
+   * Note: this works only for tables in the Hive Metastore
+   */
+  def getCatalogPartitionStats(table: Table, partitionValues: PartitionValues)(implicit session: SparkSession): Map[String, Any] = {
+    val metadata = session.sessionState.catalog.getPartition(table.tableIdentifier, partitionValues.getMapString)
+    val catalogStats = metadata.stats.map(stats =>
+      Seq(stats.rowCount.map(v => TableStatsType.NumRows.toString -> v), Some(TableStatsType.TableSizeInBytes.toString -> stats.sizeInBytes)).flatten.toMap
+    ).getOrElse(Map())
+    catalogStats + (TableStatsType.CreatedAt.toString -> metadata.createTime)
+  }
+
+  /**
+   * Note: this works only for tables in the Hive Metastore
+   */
+  def getCatalogColumnStats(table: Table)(implicit session: SparkSession): Map[String, Map[String, Any]] = {
+    session.sessionState.catalog.getTableMetadata(table.tableIdentifier).stats.toSeq.flatMap(_.colStats).toMap
+      .mapValues (
+        stats => Seq(
+          stats.distinctCount.map(ColumnStatsType.DistinctCount.toString -> _),
+          stats.nullCount.map(ColumnStatsType.NullCount.toString -> _),
+          stats.avgLen.map(ColumnStatsType.AvgLen.toString -> _),
+          stats.maxLen.map(ColumnStatsType.MaxLen.toString -> _),
+          stats.min.map(ColumnStatsType.Min.toString -> _),
+          stats.max.map(ColumnStatsType.Max.toString -> _)
+        ).flatten.toMap
+      )
+  }
+
+  /**
+   * Note: this works only for tables in the Hive Metastore
+   */
+  def getCatalogPartitionColumnStats(table: Table, partitionValues: PartitionValues)(implicit session: SparkSession): Map[String, Map[String, Any]] = {
+    session.sessionState.catalog.getPartition(table.tableIdentifier, partitionValues.getMapString).stats.toSeq.flatMap(_.colStats).toMap
+      .mapValues(
+        stats => Seq(
+          stats.distinctCount.map(ColumnStatsType.DistinctCount.toString -> _),
+          stats.nullCount.map(ColumnStatsType.NullCount.toString -> _),
+          stats.avgLen.map(ColumnStatsType.AvgLen.toString -> _),
+          stats.maxLen.map(ColumnStatsType.MaxLen.toString -> _),
+          stats.min.map(ColumnStatsType.Min.toString -> _),
+          stats.max.map(ColumnStatsType.Max.toString -> _)
+        ).flatten.toMap
+      )
+  }
+
+  /**
+   * Query partitions from catalog
+   *
+   * Note that for Hive Metastore (HMD) this might not be the best solution, as it depends on up-to-date partition metadata in HMS!
+   * We can do a directory listing for Hive tables. But for Delta Lake directory listing is not suitable, as there might be directories contain only outdated records.
```

Comment:
From the comment it looks like these methods are built in a generic way as you mention Deltalake. Should some of these methods be moved to a different file like TableUtil.scala ?

Reply:
We could do a refactoring in the future for the whole Hive thing, that's true. But currently this method is also not used for DeltaLake. I experimented with it, i think it's an interesting brick, but not yet sure if it's reliable.